euphonic/spectra.py

```diff
@@ -473,9 +527,24 @@ def _type_check(spectrum):
             assert spectrum.x_tick_labels == x_tick_labels
             y_data_magnitude[i + 1, :] = spectrum.y_data.magnitude
 
-        y_data = Quantity(y_data_magnitude, y_data_units)
+        # Process metadata
+        metadata = deepcopy(spectra[0]).metadata
+        try:
+            metadata.pop('label')
+        except (AttributeError, KeyError):
+            pass
+        for i, spectrum in enumerate(spectra):
+            if isinstance(spectrum.metadata, dict):
+                if 'label' in spectrum.metadata.keys():
+                    if not isinstance(metadata, dict):
+                        metadata = {}
+                    if not 'labels' in metadata.keys():
+                        metadata['labels'] = ['']*len(spectra)
+                    metadata['labels'][i] = spectrum.metadata['label']
```

Comment:
This took me a few tries to understand. Can we add a comment above L541 along the lines of "This is the first line with a label, initialise labels list"? At first glance it looks like L542 can clobber a list that was previously modified on L543 which would be bad.